.github/workflows/ci.yml

```diff
@@ -127,6 +127,18 @@ jobs:
           # See https://github.com/rust-lang/rust-clippy/pull/10101
           cargo clippy --all-features --all-targets --workspace -- -A clippy::derive_partial_eq_without_eq -A clippy::manual_clamp ${{ matrix.args }}
 
+      - name: Run clippy on dc/wireshark crate
+        working-directory: dc/wireshark
+        run: cargo clippy --tests
+
+      - name: Run clippy on all examples crates
+        working-directory: examples
+        run: cargo clippy --all-features --all-targets -- -A clippy::manual_clamp -A clippy::uninlined_format_args ${{ matrix.args }}
```

Comment:
Any time you use a matrix in a github action, you are spawning a  new job per matrix option. We don't want that, we want a single job for all the clippy runs. 

Also we should remove the "- toolchain: beta" matrix option. If we're not failing on it, we're not looking at it.